### ISSUE TITLE
Better error message when an immutable variable is changed

### ISSUE THREAD

**Issue by Ivo-Balbaert** [ValeLang/Vale#96]
Variables are immutable by default. If an immutable is changed, for example:

x = "world!";
x = "changed";  

The compiler returns a fairly incomprehensible message, without indicating where the error occurred:

Exception in thread "main" java.lang.RuntimeException: Assertion failed!
        at net.verdagon.vale.vfail$.apply(vassert.scala:65)
        at net.verdagon.vale.vassert$.apply(vassert.scala:30)

Better error message needed.

**Comment by Verdagon** [comment]
Thanks for the report! These are a great help for prioritizing the errors that users actually run into. 

I added an error that looks like this:
```Local named x already exists!
(If you meant to modify the variable, use the `mut` keyword beforehand.)```

(Let me know if you have any phrasing suggestions.)

To see it in action: `cat vstl/list.vale /dev/stdin | curl -X POST https://us-central1-valesite.cloudfunctions.net/vale-build -d @-` with stdin:
```fn main() export {
x = "world!";
x = "changed";
}```

Will mark this fixed when we next release. Thanks for the report!

**Comment by Verdagon** [comment]
Fixed in v0.1.1. Thanks!
